visualization/filtered_ratio_vs_purch_power.sql

```diff
@@ -0,0 +1,74 @@
+------------------------------------------
+-- Filtered to All vs Purchasing Power
+------------------------------------------
+
+WITH recent_listings AS (
+
+    -- Self join for only latest listing info gathered in the last year
+    SELECT
+        l1.listing_snapshot_key,
+        l1.standard_city,
+        l1.listing_id
+    FROM fct_listing_snapshot AS l1
+    LEFT OUTER JOIN fct_listing_snapshot AS l2
+        ON (l1.listing_id = l2.listing_id
+            AND l1.listing_report_date > l2.listing_report_date)
+    WHERE l2.listing_id IS NULL
+        AND l1.listing_report_date > dateadd(month, -12, getdate())
+
+),
+
+all_listings_count AS (
+
+    SELECT
+        count(*) AS all_list_count,
+        dim_listing.standard_city
+    FROM dim_listing
+    INNER JOIN recent_listings
+        ON recent_listings.listing_id = dim_listing.listing_id
+    GROUP BY 2
+
+),
+
+filtered_listings AS (
+
+    SELECT listing_snapshot_key
+    FROM analytics_flagged
+    WHERE 1 = 1
+        [[AND {{FREQ_BOOK_FLAG}}]]
+        [[AND {{HOME_FLAG}}]]
+        [[AND {{HIGH_AVAIL_FLAG}}]]
+        [[AND {{MULTI_HOST_FLAG}}]]
+
+),
+
+filtered_listings_count AS (
+    SELECT
+        count(DISTINCT(recent.listing_id)) AS flagged_list_count,
+        recent.standard_city
+    FROM filtered_listings
+    INNER JOIN recent_listings      AS recent
+        ON filtered_listings.listing_snapshot_key = recent.listing_snapshot_key
+    GROUP BY 2
+
+),
```

Comment:
For `all_listings_count` `filtered_listings` and `filtered_listings_count` did you consider combining into something like this:

```
SELECT
        count(DISTINCT(CASE WHEN 1=1
        [[AND {{FREQ_BOOK_FLAG}}]]
        [[AND {{HOME_FLAG}}]]
        [[AND {{HIGH_AVAIL_FLAG}}]]
        [[AND {{MULTI_HOST_FLAG}}]] THEN recent.listing_id else NULL end)) AS flagged_list_count,
        count(*) AS all_list_count,
        dim_listing.standard_city
    FROM dim_listing
    INNER JOIN recent_listings
        ON recent_listings.listing_id = dim_listing.listing_id
    INNER JOIN analytics_flagged
        ON analytics_flagged.listing_snapshot_key = recent_listings.listing_snapshot_key
    GROUP BY 3
```

This probably wouldn't work, but I figured it would be a good thing to discuss either way!

Reply:
Metabase requires the filtering variables to be in the WHERE clause, unfortunately.